### PR TITLE
[issue] enable or disable hole dynamically.

### LINKED ISSUE #20: The OverlayTutorialHole.enabled does not work
The OverlayTutorialHole.enabled does not work as advertised.

This is due to the `enabled` state is not handled properly.

### PR DESCRIPTION
**Pull request by Guang1234567**
Hello dev:

Thanks your great job.  

And the PR  as title said, thanks. 

BTW, maybe `OverlayTutorialController` better than `final overlayTutorialScopeState =
        context.findAncestorStateOfType<_OverlayTutorialScopeState>();`

### CODE REVIEW

File: lib/src/widgets/overlay_tutorial_hole.dart [hunk 1]
```diff
@@ -24,20 +24,58 @@ class OverlayTutorialHole extends StatefulWidget {
...
+    super.didUpdateWidget(oldWidget);
+    final oldState = oldWidget.enabled;
+    final currentState = widget.enabled;
+    if (oldState != currentState) {
+      final overlayTutorialScope = _overlayTutorialScopeState;
+      if (overlayTutorialScope != null) {
+        if (currentState) {
+          overlayTutorialScope._overlayTutorialHoles[widget] = context;
+        } else {
+          //overlayTutorialScope._overlayTutorialHoles.remove(widget);
```
Comment (TabooSun) [inline lib/src/widgets/overlay_tutorial_hole.dart]: Unused comment should be removed.

File: lib/src/widgets/overlay_tutorial_scope.dart [hunk 2]
```diff
@@ -69,27 +69,29 @@ class _OverlayTutorialScopeState extends State<OverlayTutorialScope> {
...
+                      left: position.dx,
+                      top: position.dy,
+                      child: hint.builder(
+                        context,
+                        entryRect,
+                        rRect,
+                      ),
+                    );
+                  } else {
+                    return SizedBox(width: 0, height: 0);
```
Comment (TabooSun) [inline lib/src/widgets/overlay_tutorial_scope.dart]: I would suggest changing to `const SizedBox.shrink()`. This is due to it's more readable.

**Review by TabooSun — APPROVED** [review]

**Comment by TabooSun** [comment]
@Guang1234567 Thanks for fixing this bug. I will publish it later.

File: lib/src/widgets/overlay_tutorial_scope.dart (hunk 2 above)
Comment (Guang1234567) [reply lib/src/widgets/overlay_tutorial_scope.dart]: > I would suggest changing to `const SizedBox.shrink()`. This is due to it's more readable.

Good !

**Pull request merged**